lint_rules/lint_rules/debug_rules.py

```diff
@@ -284,13 +284,25 @@ def fix_subroutine(cls, subroutine, rule_report, config):
                 vtype = arg.type.clone(shape=new_shape, scope=subroutine)
                 new_vars += as_tuple(arg.clone(type=vtype, dimensions=new_shape, scope=subroutine))
 
-        #TODO: add 'VariableDeclaration.symbols' should be of type 'Variable' rather than 'Expression'
+        # simplify variable declarations
+        single_variable_declaration(subroutine)
+
+        #TODO: 'VariableDeclaration.symbols' should be of type 'Variable' rather than 'Expression'
         # to enable case-insensitive search here
         new_var_names = [v.name.lower() for v in new_vars]
-        subroutine.variables = [var for var in subroutine.variables if not var.name.lower() in new_var_names]
 
-        subroutine.body = Transformer(node_map).visit(subroutine.body)
-        subroutine.variables += new_vars
+        routine = subroutine.clone()
```

Comment:
This seems pretty invasive as a change, given none of the expression manipulation below seems to deal with source strings. Is this strictly needed, or should this be in a separate PR?

Reply:
Yes in hindsight, piggy-backing `DebugRule` fixes onto this PR was not the wisest move. I'll remove those commits, and add them to some other `DebugRule` fixes that I plan on PR'ing soon.